### ISSUE TITLE
Add a message validation server.

### ISSUE THREAD

**Issue by FellowTraveler** [FellowTraveler/Open-Transactions-old#28]
There should be a front server that performs certain message validation, for security reasons.

VoucherSafe seems to have a front server that manages the user accounts, and a "voucher publisher" server where the tokens are signed and exchanged.  (OT does this in a single server.)

THAT's when I realized, couldn't OT work in that exact fashion???

Couldn't I put up an OT server that ONLY issues notes, and another OT server that ONLY manages accounts?

and have one manage all the users, and have the other process certain messages on behalf of the first?

in effect, just putting up two copies of OT — and replicating the VoucherSafe system?
It would still require a modification to store the actual serial numbered notes, but I think that's about it...

Basically look at my Fully-Anonymous diagram.  The "cash-only" interface would be the first OT server, managing the accounts, and then the other OT server would be the main server.

then add the modification for tracking serial numbers, and it seems the same as Voucher Safe, no?

The key is not in the technical setup itself — it's more of a configuration issue — the KEY is in how you structure the entities, just like you would structure corporate entities. Same science.
